spec/services/reinstate_pending_conditions_spec.rb

```diff
@@ -65,4 +65,16 @@
   describe 'validations' do
     include_examples 'confirm deferred offer validations', :reinstate_pending_conditions
   end
+
+  context 'when sending the email' do
+    it 'sends reinstated offer email with correct content', :sidekiq do
+      new_course_option.course.update!(start_date: original_course.start_date + 1.year)
+
+      expect {
+        service.save!
+      }.to change { ActionMailer::Base.deliveries.count }.by(1)
+
+      expect(ActionMailer::Base.deliveries.first.body.raw_source).to include(new_course_option.course.start_date.to_fs(:month_and_year))
+    end
+  end
```

Comment:
When I wrote this spec, before the fix, this spec passed many times and failed some due to the nature of the commit happening fast in my local setup. But that's the best I could reproduce the transactional issue described on the PR. Let me know if you are cool with it or have any other idea!